File: lib_local/gallery/src/main/java/com/yanzhenjie/album/api/camera/AlbumCamera.java
package com.yanzhenjie.album.api.camera;

import android.content.Context;

import com.yanzhenjie.album.api.ImageCameraWrapper;
import com.yanzhenjie.album.api.VideoCameraWrapper;

/**
 * 相机总入口类
 * 实现 Camera 接口，提供【拍照】和【录像】两大功能
 */
public class AlbumCamera implements Camera<ImageCameraWrapper, VideoCameraWrapper> {
    private final Context mContext;

    public AlbumCamera(Context context) {
        mContext = context;
    }

    /**
     * 打开相机：只拍照
     */
    @Override
    public ImageCameraWrapper image() {
        return new ImageCameraWrapper(mContext);
    }

    /**
     * 打开相机：只录像
     */
    @Override
    public VideoCameraWrapper video() {
        return new VideoCameraWrapper(mContext);
    }

}

File: lib_local/gallery/src/main/java/com/yanzhenjie/album/api/camera/AlbumCamera.kt
package com.yanzhenjie.album.api.camera

import android.content.Context
import com.yanzhenjie.album.api.ImageCameraWrapper
import com.yanzhenjie.album.api.VideoCameraWrapper

/**
 * Created by YanZhenjie on 2017/8/18.
 */
class AlbumCamera(private val mContext: Context) : Camera<ImageCameraWrapper, VideoCameraWrapper> {

    override fun image(): ImageCameraWrapper {
        return ImageCameraWrapper(mContext)
    }

    override fun video(): VideoCameraWrapper {
        return VideoCameraWrapper(mContext)
    }

}